modules/apps/users-admin/users-admin-test/src/testIntegration/java/com/liferay/users/admin/web/internal/portlet/action/test/EditUserMVCActionCommandTest.java

```diff
@@ -0,0 +1,273 @@
+ */
+@RunWith(Arquillian.class)
+@Sync
+public class EditUserMVCActionCommandTest {
+
+	@ClassRule
+	@Rule
+	public static final AggregateTestRule aggregateTestRule =
+		new AggregateTestRule(
+			new LiferayIntegrationTestRule(),
+			PermissionCheckerMethodTestRule.INSTANCE);
+
+	@Test
+	public void testUpdatePasswordDoesNotNotifyUnapprovedUser()
+		throws Exception {
+
+		String originalName = PrincipalThreadLocal.getName();
+
+		PermissionChecker originalPermissionChecker =
+			PermissionThreadLocal.getPermissionChecker();
+
+		try {
+			PrincipalThreadLocal.setName(TestPropsValues.getUserId());
+
+			PermissionChecker adminPermissionChecker =
+				PermissionCheckerFactoryUtil.create(TestPropsValues.getUser());
+
+			PermissionThreadLocal.setPermissionChecker(adminPermissionChecker);
+
+			_workflowDefinitionLinkLocalService.addWorkflowDefinitionLink(
+				null, TestPropsValues.getUserId(),
+				TestPropsValues.getCompanyId(),
+				GroupConstants.DEFAULT_LIVE_GROUP_ID, User.class.getName(), 0,
+				0, "Single Approver", 1);
+
+			User user = _userLocalService.addUserWithWorkflow(
+				0, TestPropsValues.getCompanyId(), false, "test", "test", false,
+				RandomTestUtil.randomString(),
+				RandomTestUtil.randomString() + "@liferay.com", LocaleUtil.US,
+				RandomTestUtil.randomString(), RandomTestUtil.randomString(),
+				RandomTestUtil.randomString(), 0, 0, true, 1, 1, 1970,
+				StringPool.BLANK, UserConstants.TYPE_REGULAR, null, null, null,
+				null, true,
+				ServiceContextTestUtil.getServiceContext(
+					TestPropsValues.getCompanyId(),
+					TestPropsValues.getGroupId(), TestPropsValues.getUserId()));
+
+			Assert.assertEquals(
+				WorkflowConstants.STATUS_PENDING, user.getStatus());
+
+			_processActionRequestInPublication(
+				"/users_admin/edit_user",
+				HashMapBuilder.put(
+					Constants.CMD, Constants.RESTORE
+				).put(
+					"deleteUserIds", String.valueOf(user.getUserId())
+				).build(),
+				adminPermissionChecker);
+
+			List<WorkflowTask> workflowTasks =
+				_workflowTaskManager.getWorkflowTasksByWorkflowInstance(
+					TestPropsValues.getCompanyId(), null,
+					_workflowInstanceLinkLocalService.fetchWorkflowInstanceLink(
+						TestPropsValues.getCompanyId(),
+						WorkflowConstants.DEFAULT_GROUP_ID,
+						User.class.getName(), user.getUserId()
+					).getWorkflowInstanceId(),
+					null, QueryUtil.ALL_POS, QueryUtil.ALL_POS, null);
+
+			Assert.assertEquals(
+				workflowTasks.toString(), 1, workflowTasks.size());
+
+			WorkflowTask workflowTask = workflowTasks.get(0);
+
+			Assert.assertTrue(workflowTask.isCompleted());
+
+			user = _userLocalService.getUser(user.getUserId());
+
+			Assert.assertEquals(
+				WorkflowConstants.STATUS_APPROVED, user.getStatus());
+		}
+		finally {
+			PrincipalThreadLocal.setName(originalName);
+
+			PermissionThreadLocal.setPermissionChecker(
+				originalPermissionChecker);
+		}
+	}
+
+	private ThemeDisplay _getThemeDisplay(PermissionChecker permissionChecker)
```

Comment:
I would have put this in a setUp method, but I see this is used in other places too, so it's fine